sdk/src/main/java/com/mapbox/maps/MapboxMap.kt

```diff
@@ -169,18 +170,40 @@ class MapboxMap internal constructor(
     onStyleLoaded: Style.OnStyleLoaded? = null
   ) {
     this.style = style
-    styleExtension.images.forEach {
-      it.bindTo(style)
-    }
+    var resourceCount = styleExtension.resourceCount
     styleExtension.sources.forEach {
+      if (it is GeoJsonSource) {
+        it.bindTo(style)
+        it.addOnGeoJsonParsedListener {
+          resourceCount--
```

Comment:
shall we remove the `OnGeoJsonParsedListener` after the parsing is finished?

Reply:
in case the source will be modified later by users, and this block will still be called.

Reply:
I don't think it's that important in this case tbh. We can't do it right now because of there's no direct reference to anonymous lambda.

Reply:
OK, I'll remove it, you're correct.